src/universal/modules/outcomeCard/containers/OutcomeCard/OutcomeCardContainer.js

```diff
@@ -72,6 +85,19 @@ class OutcomeCardContainer extends Component {
     });
   };
 
+  isEditing = () => {
+    return !this.state.activeEditingClients.isEmpty()
+  }
+
+  trackEditingClient = (uid, isEditing) => {
+    this.setState((curState) => {
```

Comment:
Oh that's an interesting problem! A couple options pop up into my head initially (may think of more in a bit)

- I'm not actually sure which event the browser will fire first - the `focusout` event on the editor or the `click` event on the menu toggle. If that click event occurs first (not sure if it does), then we can add the menu to the set at that point rather than waiting for `componentWillMount` on the menu.

- This one's is a bit of a hack, but it could lead to a possibly less twitchy UX in general: we could keep track of a timer within the container's state in order to buffer changes to the edit state. That way, within any given `m` milliseconds after the edit client set empties, we'll still consider the component to be "edited". This would generally handle the cases in which you stop editing something only to start editing it right again. We could probably make use of a higher-order component when it comes to implementation so that other live components can opt in to this behavior. This also has the nice property of keeping the backend from having to handle trivial edit state mutations.